### DISCUSSION TITLE
Why is `meld` marked as "discontinued"?

### DISCUSSION THREAD

**Discussion by alexreg** [Homebrew/discussions#5473] [Casks] [answered]
### Output of `brew config`

```shell
N.A.
```

### Output of `brew doctor`

```shell
N.A.
```

### Description of issue

Why is `meld` marked as "discontinued"?

### Relevant casks

meld

**Comment by p-linnane** [comment, marked as answer]
It hasn't had any releases in over 4 years and is only built for x86_64. If they decide to cut new releases, we can remove that and bump the version.

**Reply by alexreg** [reply]
Okay, that's fair enough. Thanks for clarifying. (I still use it because there's nothing better on macOS sadly.)

**Comment by DrStrangepork** [comment]
Now that there is a different active source at https://gitlab.com/dehesselle/meld_macos, why can't we just switch the cask to point to the active project?

**Reply by p-linnane** [reply]
Unless the original upstream blesses this repo we won't be able to use it. Switching to a different source is not something existing users would be aware of, and it causes potential namejacking issues.

**Reply by alexreg** [reply]
As at least a temporary solution, I've created a cask for this fork.
```shell
brew install alexreg/cask-extra/meld
```

**Reply by dbarnett** [reply]
Thank you for this!! The other version was constantly crashing where yours is working beautifully!

**Reply by alexreg** [reply]
@dbarnett No problem. Now this same app is in homebrew-cask though, as `dehesselle-meld`, so we should probably use that going forwards. :-)

**Comment by rembertomitimiti** [comment]
@p-linnane It now appears active.

Is this enough to reactivate the cask?

Release notes
2025-01-08 meld 3.22.3

https://gitlab.gnome.org/GNOME/meld/-/releases

**Reply by p-linnane** [reply]
No, since that is not the same upstream. Unless the original upstream points to a designated successor, we won't change the cask. There is another meld fork mentioned in this thread.

**Reply by mhnestler** [reply]
Is the key fact here that https://meldmerge.org/ links to https://github.com/yousseb/meld/releases, thus "endorsing" yousseb's fork? The text on the homepage is:

> Meld is not _yet_ supported on OS X. Third-party binaries [are available](https://github.com/yousseb/meld/releases), and you can install from [Homebrew](https://brew.sh/), [MacPorts](https://www.macports.org/) or [Fink](http://www.finkproject.org/).

So if https://meldmerge.org/ updated its recommendation, that should be authoritative, right?

BTW, the [release](https://gitlab.gnome.org/GNOME/meld/-/releases#2025-03-23-meld-3230) 3 weeks ago (2025-03-23, version 3.23.0) [contains](https://gitlab.gnome.org/GNOME/meld/-/merge_requests/124) work from **dehesselle** going [towards upstream](https://gitlab.gnome.org/GNOME/meld/-/issues/804#note_2298525) macOS support. But the extent of that (AFAIK) is that upstream plans to offer CI-built binaries. Still, @kaiw could probably update their homepage since they're working with @dehesselle directly.

**Reply by p-linnane** [reply]
> Is the key fact here that [meldmerge.org](https://meldmerge.org/) links to [yousseb/meld/releases](https://github.com/yousseb/meld/releases), thus "endorsing" yousseb's fork?

No. The cask in question doesn't even reference meldmerge.org:
```
  url "https://github.com/yousseb/meld/releases/download/osx-#{version.csv.second}/meldmerge.dmg",
      verified: "github.com/yousseb/meld/"
  name "Meld for OSX"
  desc "Visual diff and merge tool"
  homepage "https://yousseb.github.io/meld/"
```

That being said, if meldmerge.org points to @dehesselle's fork as the recommendation, we can make that the main `meld` cask. Ping me if this happens and I'll open the necessary PR.

**Reply by dehesselle** [reply]
> Is the key fact here that https://meldmerge.org/ links to https://github.com/yousseb/meld/releases, thus "endorsing" yousseb's fork?

No, this just hasn't been updated in a while. It probably makes sense to redirect this to my fork until upstream can fully stand on its own feet. (limited resources - progress is slower than anticipated)
I'll ask.

> But the extent of that (AFAIK) is that upstream plans to offer CI-built binaries.

That's what I bring to the table, yes. But I'm hoping that someone else shows up to help upstreaming yousseb's macOS specific modifications (e..g. native window decorations).

**Reply by kaiw** [reply]
I've just update the main Meld page to point to dehesselle's builds. At some point we should hopefully have those as regular parts of Meld releases.

For what it's worth, while I maintain the site at both https://meldmerge.org and https://meld.app, the former is the legacy URL.

**Reply by p-linnane** [reply]
Thanks for the update. I've opened https://github.com/Homebrew/homebrew-cask/pull/209550 to make this change.
